### PR TITLE
Implement custom comparison for ControlledSequence

### PR DESCRIPTION
**Pull request by EmilianoG-byte**
**Context:**

Currently, `qml.equal` does not have a registered function that compares two `ControlledSequence` objects correctly. This PR addresses this issue.

**Description of the Change:**

Registers a new overloaded implementation of `_equal` for arguments of type `ControlledSequence`. For two objects of this type, `qml.equal` compares if:

- their control properties are the same
- their base's are the same to the requested tolerances and parameter checks.

For example:

``` python
op1 = qml.ControlledSequence(qml.PauliX(0), control=1)
op2 = qml.ControlledSequence(qml.PauliX(0), control=1)
qml.equal(op1, op2)

>>> True
```

**Benefits:**

Correct comparison of two `ControlledSequence` objects is now possible.

**Possible Drawbacks:**

None.

**Related GitHub Issues:**

Closes: #4797

### CODE REVIEW

**Comment by EmilianoG-byte** [comment]
Hi!

I have based this PR on an older commit 970d30c252c148f2bcf8ff14793f6ad349b55cc7 (from 26.10) since the newest commit was throwing some import errors when I pulled from master, and this was one of the last ones were the CI checks were passing. I see the one only conflict is with the changelog, so hopefully doing this way was ok :)

Let me know what you think about my changes!

**Comment by rmoyard** [comment]
Hello @EmilianoG-byte , thanks for your contribution! I will give an initial review of your PR this afternoon.

File: tests/ops/functions/test_equal.py [hunk 1]
```diff
@@ -1399,6 +1452,15 @@ def test_control_wires_comparison(self, wire1, wire2, res):
...
+        """Test that equal compares control for ControlledSequence operators"""
+        base1 = qml.Hadamard(0)
+        base2 = qml.Hadamard(0)
+        op1 = ControlledSequence(base1, control=wires1)
+        op2 = ControlledSequence(base2, control=wires2)
+        assert qml.equal(op1, op2) == res
+
     @pytest.mark.parametrize(("wire1", "wire2", "res"), WIRES)
     def test_controlled_work_wires_comparison(self, wire1, wire2, res):
         """Test that equal compares work_wires for Controlled operators"""
```
Comment (rmoyard) [inline tests/ops/functions/test_equal.py:1466]: Great testing!

**Review by lillian542 — APPROVED** [review]
This looks good @EmilianoG-byte, thanks for the thorough tests! 🚀

**Comment by EmilianoG-byte** [comment]
> This looks good @EmilianoG-byte, thanks for the thorough tests! 🚀

Thank you 😃 @lillian542 !!

**Comment by EmilianoG-byte** [comment]
Hi @rmoyard, I made the changes we discussed 😊
Let me know if some of them were still up for discussion or if they are okay!

**Review by rmoyard — APPROVED** [review]
Thanks for the PR :100:

**Comment by rmoyard** [comment]
Hi @EmilianoG-byte, I am glad you solved the issue! For the second part, in general you are welcome to use it but I also like when it is tested separated. It makes it easier to refactor if needed.

**Comment by Alex-Preciado** [comment]
Awesome! Thank you so much @EmilianoG-byte. Merging this now 😎

**Comment by EmilianoG-byte** [comment]
> Awesome! Thank you so much @EmilianoG-byte. Merging this now 😎

Thank you @Alex-Preciado 😄

**Pull request merged**